nixos/modules/services/audio/wyoming/faster-whisper.nix

```diff
@@ -0,0 +1,162 @@
+    package = mkPackageOptionMD pkgs "wyoming-faster-whisper" { };
+
+    servers = mkOption {
+      default = {};
+      description = mdDoc ''
+        Attribute set of faster-whisper instances to spawn.
+      '';
+      type = types.attrsOf (types.submodule (
+        { ... }: {
+          options = {
+            enable = mkEnableOption (mdDoc "Wyoming faster-whisper server");
+
+
+            model = mkOption {
+              type = types.enum [
+                "tiny"
+                "tiny-int8"
+                "base"
+                "base-int8"
+                "small"
+                "small-int8"
+                "medium"
+                "medium-int8"
+              ];
+              default = "tiny-int8";
+              example = "medium-int8";
+              description = mdDoc ''
+                Name of the voice model to use. See the following website for samples:
+                https://rhasspy.github.io/faster-whisper-samples/
+              '';
+            };
+
+            uri = mkOption {
+              type = strMatching "^(tcp|unix)://.*$";
+              example = "tcp://0.0.0.0:10300";
+              description = mdDoc ''
+                URI to bind the wyoming server to.
+              '';
+            };
+
+            device = mkOption {
+              # https://opennmt.net/CTranslate2/python/ctranslate2.models.Whisper.html#
+              type = types.enum [
+                "cpu"
+                "cuda"
+                "auto"
+              ];
+              default = 0;
+              description = mdDoc ''
+                Id of a speaker in a multi-speaker model.
+              '';
+            };
+
+            extraArgs = mkOption {
+              type = listOf str;
+              default = [ ];
+              description = mdDoc ''
+                Extra arguments to pass to the server commandline.
+              '';
+              apply = escapeShellArgs;
+            };
+          };
+        }
+      ));
+    };
+  };
+
+  config = let
+    inherit (lib)
+      mapAttrs'
+      mkIf
+      nameValuePair
+    ;
+  in mkIf (cfg.servers != {}) {
+    systemd.services = mapAttrs' (server: options:
+      nameValuePair "wyoming-faster-whisper-${server}" {
+        description = "Wyoming faster-whisper server instance ${server}";
+        after = [
+          "network-online.target"
+        ];
+        wantedBy = [
+          "multi-user.target"
+        ];
+        serviceConfig = {
+          DynamicUser = true;
+          User = "wyoming-faster-whisper";
+          StateDirectory = "wyoming/faster-whisper";
+          # https://github.com/home-assistant/addons/blob/master/whisper/rootfs/etc/s6-overlay/s6-rc.d/whisper/run
+          ExecStart = ''
+            ${cfg.package}/bin/wyoming-faster-whisper \
+              --data-dir $STATE_DIRECTORY \
+              --download-dir $STATE_DIRECTORY \
+              --model ${options.model} \
+              --language ${options.language} \
```

Comment:
The `options.language` option is not defined. Should be something like:
```nix
language = mkOption {
  type = types.enum [
    "auto"
    "af"
    "am"
    "ar"
    "as"
    "az"
    "ba"
    "be"
    "bg"
    "bn"
    "bo"
    "br"
    "bs"
    "ca"
    "cs"
    "cy"
    "da"
    "de"
    "el"
    "en"
    "es"
    "et"
    "eu"
    "fa"
    "fi"
    "fo"
    "fr"
    "gl"
    "gu"
    "ha"
    "haw"
    "he"
    "hi"
    "hr"
    "ht"
    "hu"
    "hy"
    "id"
    "is"
    "it"
    "ja"
    "jw"
    "ka"
    "kk"
    "km"
    "kn"
    "ko"
    "la"
    "lb"
    "ln"
    "lo"
    "lt"
    "lv"
    "mg"
    "mi"
    "mk"
    "ml"
    "mn"
    "mr"
    "ms"
    "mt"
    "my"
    "ne"
    "nl"
    "nn"
    "no"
    "oc"
    "pa"
    "pl"
    "ps"
    "pt"
    "ro"
    "ru"
    "sa"
    "sd"
    "si"
    "sk"
    "sl"
    "sn"
    "so"
    "sq"
    "sr"
    "su"
    "sv"
    "sw"
    "ta"
    "te"
    "tg"
    "th"
    "tk"
    "tl"
    "tr"
    "tt"
    "uk"
    "ur"
    "uz"
    "vi"
    "yi"
    "yo"
    "zh"
  ];
  default = "auto";
  example = "en";
};
```

From https://github.com/home-assistant/addons/blob/master/whisper/config.yaml#L20